drivers/core_demographics_report/app/models/core_demographics_report/details_column.rb

```diff
@@ -0,0 +1,46 @@
+# CoreDemographicsReport::DetailsColumn
+module CoreDemographicsReport
+  DetailsColumn = Struct.new(:label, :index, :user, :project_id_index, keyword_init: true) do
+    def value(row)
+      raw_value = row[index]
+
+      project_id = row[project_id_index]
+      policy = user.policies.for_project(project_id)
+      pii_value(raw_value, policy)
+    end
+
+    protected
+
+    REDACTED = 'Redacted'.freeze
+    def pii_value(raw_value, policy)
+      case label
+      when 'First Name', 'Last Name'
+        policy.can_view_client_name? ? raw_value : REDACTED
```

Comment:
This should probably be something like `policy.name_or_redacted(:first_name, :can_view_client_name?)`

Reply:
I don't think display logic should go in the policy object (separation of concerns and all that).
It could be rolled up into PiiProvider, maybe a class method would work for this case where we are iterating over cells 
`PiiProvider.first_name(raw_value, policy: policy)`